JavaScript/Topics/REST_APIs/XMLHttpRequest/README.md

```diff
@@ -0,0 +1,105 @@
+# XMLHttpRequest
+---
+
+In this Tutorial , we will learn about  **XMLHttpRequest**.
+
+## What is  XMLHttpRequest?
+
+The XMLHttpRequest allows us to request data from servers. You can retrieve the data from a *URL* without refreshing the full page.
+
+You will think that XMLHttpRequest can retrieve only XML type data because of its name but despite its name , it can be used to retrieve any type of data.
+
+
+## How to send a Request?
+
+```js
+var xhr = new XMLHttpRequest();
+ ```
+
+1. An instance of the XMLHttpRequest object is created using object constructor. It uses the new keyword and stores the object in a variable. The variable name xhr is short for  XMLHttpRequest(the name of the object).
+
+```js
+xhr.open('GET','data/test.json',true);
+```     
+
+2. The XMLHttpRequest object's open () method prepares the request. It has three parameters:
+i) The HTTP method
+ii) The url of the page that will handle your request.(Here, "test.json" file used in the example above).
+iii) A Boolean indicating if it should be asynchronous.
+
+
+```js
+xhr.send(null);
+```
+3. The send() method is the one that sends the prepared request to the server. Extra information ccan be passed to the server in the parentheses. If no extra information is sent, you may see the keyword null used (although it is not strictly needed).
+
+
+## How to handle a Response?
+
+
+```js
+     xhr.onload = function()
+     {
+       if(xhr.status === 200){
+         // code to process the results from server
+       }     
+     }
+ ```
+1. When the browser has received and loaded a response from the server, the on load event will fire. This will trigger a function (here, it is an anonymous function).
```

Comment:
```suggestion
1. When the browser has received and loaded a response from the server, the onLoad event will fire. This will trigger a function (here, it is an anonymous function).
```

Reply:
The required changes has been done.